src/box/index.cc

```diff
@@ -457,23 +458,58 @@ box_index_iterator(uint32_t space_id, uint32_t index_id, int type,
 	uint32_t part_count = mp_decode_array(&key);
 	if (key_validate(index->def, itype, key, part_count))
 		return NULL;
+	const char *pos, *pos_end;
+	char *buf = NULL;
+	if (packed_pos != NULL && packed_pos != packed_pos_end) {
+		uint32_t buf_size =
+			iterator_position_unpack_bufsize(packed_pos,
+							 packed_pos_end);
+		buf = (char *)xmalloc(buf_size);
```

Comment:
Let's allocate the data from `runtime_alloc` - it should be much faster than `malloc`.

Reply:
Done - introduced `tuple_runtime_memory_alloc(free)` methods right in pagination commit.